common/events/events_io.cpp

```diff
@@ -123,6 +123,32 @@ void AsyncWriterFromWriter::Cancel() {
 	}
 }
 
+ReaderFromAsyncReader::ReaderFromAsyncReader(AsyncReaderFromEventLoopFunc func) {
```

Comment:
So this magic is to avoid implementation of a synchronous reader for what we already have an async reader for?